moveit_core/collision_detection_bullet/include/moveit/collision_detection_bullet/bullet_integration/bullet_utils.h

```diff
@@ -345,7 +447,107 @@ inline btScalar addDiscreteSingleResult(btManifoldPoint& cp, const btCollisionOb
   return 1;
 }
 
-/** \brief Processes a contact point */
+inline btScalar addCastSingleResult(btManifoldPoint& cp, const btCollisionObjectWrapper* colObj0Wrap, int /*index0*/,
+                                    const btCollisionObjectWrapper* colObj1Wrap, int /*index1*/,
+                                    ContactTestData& collisions)
+{
+  assert(dynamic_cast<const CollisionObjectWrapper*>(colObj0Wrap->getCollisionObject()) != nullptr);
+  assert(dynamic_cast<const CollisionObjectWrapper*>(colObj1Wrap->getCollisionObject()) != nullptr);
+
+  const CollisionObjectWrapper* cd0 = static_cast<const CollisionObjectWrapper*>(colObj0Wrap->getCollisionObject());
+  const CollisionObjectWrapper* cd1 = static_cast<const CollisionObjectWrapper*>(colObj1Wrap->getCollisionObject());
+
+  std::pair<std::string, std::string> pc = getObjectPairKey(cd0->getName(), cd1->getName());
+
+  const auto& it = collisions.res.contacts.find(pc);
+  bool found = (it != collisions.res.contacts.end());
+
+  collision_detection::Contact contact;
+  contact.body_name_1 = cd0->getName();
+  contact.body_name_2 = cd1->getName();
+  contact.depth = static_cast<double>(cp.m_distance1);
+  contact.normal = convertBtToEigen(-1 * cp.m_normalWorldOnB);
+  contact.pos = convertBtToEigen(cp.m_positionWorldOnA);
+
+  contact.body_type_1 = cd0->getTypeID();
+  contact.body_type_2 = cd0->getTypeID();
+
+  collision_detection::Contact* col = processResult(collisions, contact, pc, found);
+
+  if (!col)
+  {
+    return 0;
+  }
+
+  assert(!((cd0->m_collisionFilterGroup == btBroadphaseProxy::KinematicFilter) &&
+           (cd1->m_collisionFilterGroup == btBroadphaseProxy::KinematicFilter)));
+
+  bool cast_shape_is_first = cd0->m_collisionFilterGroup == btBroadphaseProxy::KinematicFilter;
+
+  btVector3 normal_world_from_cast = -(cast_shape_is_first ? 1 : -1) * cp.m_normalWorldOnB;
```

Comment:
can we simplify this expression?: `(cast_shape_is_first ? -1 : 1) * ...`.

Reply:
sure, done.